src/usdhc.rs

```diff
@@ -0,0 +1,83 @@
+//! USDHC pad configuration
+
+/// Type tag for the reset pin
+pub enum Cmd {}
+/// Type tag for the reset pin
+pub enum Clk {}
+/// Type tag for the reset pin
+pub enum Wp {}
+/// Type tag for the reset pin
+pub enum CdB {}
+/// Type tag for the data0 pin
+pub enum Data0 {}
+/// Type tag for the data1 pin
+pub enum Data1 {}
+/// Type tag for the data2 pin
+pub enum Data2 {}
+/// Type tag for the data3 pin
+pub enum Data3 {}
+
+/// A uSDHC pin signal
+mod private {
+    pub trait Signal {}
+    impl Signal for super::Cmd {}
+    impl Signal for super::Clk {}
+    impl Signal for super::Wp {}
+    impl Signal for super::CdB {}
+    impl Signal for super::Data0 {}
+    impl Signal for super::Data1 {}
+    impl Signal for super::Data2 {}
+    impl Signal for super::Data3 {}
+}
```

Comment:
Combining the [sealed pattern](https://rust-lang.github.io/api-guidelines/future-proofing.html#future-proofing) with the constraint for a pin signal means that users studying the documentation cannot see the set of uSDHC signals. `usdhc::Signal` is not exposed, so cannot be selected in docs.

Contrast that with SAI pins, which separates the seal from the signal constraint. This lets users select `Signal`, and see all implementations (depicted below).

Is there a reason that uSDHC traits take a different approach?

<img width="393" alt="image" src="https://user-images.githubusercontent.com/7933926/148685591-3276a2bc-996e-4fb8-be59-e446a93f8903.png">